packages/protocol/contracts/governance/Governance.sol

```diff
@@ -1252,6 +1255,43 @@ contract Governance is
     }
   }
 
+  /**
+   * @notice Removes the proposal from the queue if `lastDequeue` time has passed.
+   * @param proposalId The ID of the proposal.
+   * @dev If any of the top proposals have expired, they are deleted.
+   */
+  function dequeueProposalIfReady(uint256 proposalId) public returns (bool isProposalDequeued) {
+    isProposalDequeued = false;
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= lastDequeue.add(dequeueFrequency)) {
+      Proposals.Proposal storage proposal = proposals[proposalId];
```

Comment:
It seems to me that we allow to dequeue proposals that are expired. Or is there any reason why we are missign there this check

` if (_isQueuedProposalExpired(proposal)) {
          emit ProposalExpired(proposalId);
          continue;
        }`

Reply:
great point, tbh quite some legacy going on in that section - the @dev comment specifically mentions that such proposals are deleted - not familiar with the decisions but it appears to be intentionally there for a reason - thus the logic with the emptyIndices further down below.